spring-integration-file/src/main/java/org/springframework/integration/file/remote/gateway/AbstractRemoteFileOutboundGateway.java

```diff
@@ -187,21 +187,27 @@ public AbstractRemoteFileOutboundGateway(RemoteFileTemplate<F> remoteFileTemplat
 	 * 'get' etc), and an expression to determine the filename.
 	 * @param remoteFileTemplate the remote file template.
 	 * @param command the command.
-	 * @param expression the filename expression.
+	 * @param expressionArg the filename expression.
 	 */
 	public AbstractRemoteFileOutboundGateway(RemoteFileTemplate<F> remoteFileTemplate, Command command,
-			@Nullable String expression) {
+			@Nullable String expressionArg) {
 
 		Assert.notNull(remoteFileTemplate, "'remoteFileTemplate' cannot be null");
 		this.remoteFileTemplate = remoteFileTemplate;
 		this.command = command;
+		String expression = expressionArg;
+		boolean noExpressionNeeded = Command.LS.equals(this.command)
+				|| Command.NLST.equals(this.command)
+				|| Command.PUT.equals(this.command)
+				|| Command.MPUT.equals(this.command);
+		if ("++xsd.expression.default++".equals(expression)) {
+			expression = noExpressionNeeded ? null : "payload";
```

Comment:
Why can't we set it to `null` from the parser and override it to `"payload"` if needed according command logic?